pkg/apis/core/v1/conversion.go

```diff
@@ -555,3 +563,109 @@ func Convert_core_PersistentVolumeSpec_To_v1_PersistentVolumeSpec(in *core.Persi
 func Convert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in *v1.PersistentVolumeSpec, out *core.PersistentVolumeSpec, s conversion.Scope) error {
 	return autoConvert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in, out, s)
 }
+
+// handleSeccompVersionSkew implements the version skew behavior described in:
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190717-seccomp-ga.md#version-skew-strategy
+func handleSeccompVersionSkew(in *core.Pod, out *v1.Pod) error {
+	klog.V(4).Infof("Handling seccomp version skew fro pod %v", in)
+	annotation, annotationAvailable := in.Annotations[v1.SeccompPodAnnotationKey]
+
+	var field *core.SeccompProfile
+	if in.Spec.SecurityContext != nil {
+		field = in.Spec.SecurityContext.SeccompProfile
+	}
+
+	// If no seccomp annotations or fields are specified, then no further
+	// action is necessary.
+	if field == nil && !annotationAvailable {
+		return nil
+	}
+
+	// TODO: change those to global constants
+	const (
+		profileUnconfined = "unconfined"
+		localhostPrefix   = "localhost/"
+	)
+
+	// If only seccomp fields are specified, add the corresponding annotations.
+	// This ensures that the fields are enforced even if the node version
+	// trails the API version
+	if field != nil && !annotationAvailable {
+		switch field.Type {
+		case core.SeccompProfileUnconfined:
+			out.Annotations[v1.SeccompPodAnnotationKey] = profileUnconfined
+
+		case core.SeccompProfileTypeDefault:
+			out.Annotations[v1.SeccompPodAnnotationKey] = v1.SeccompProfileRuntimeDefault
+
+		case core.SeccompProfileLocalhost:
+			out.Annotations[v1.SeccompPodAnnotationKey] = localhostPrefix + *field.LocalhostProfile
+		}
+		klog.V(4).Infof(
+			"Added seccomp annotation %s=%s",
+			v1.SeccompPodAnnotationKey,
+			out.Annotations[v1.SeccompPodAnnotationKey],
+		)
+	}
+
+	// If only seccomp annotations are specified, copy the values into the
+	// corresponding fields. This ensures that existing applications continue
+	// to enforce seccomp, and prevents the kubelet from needing to resolve
+	// annotations & fields.
+	if field == nil && annotationAvailable {
+		// Ensure that the security context is not nil
+		if out.Spec.SecurityContext == nil {
+			out.Spec.SecurityContext = &v1.PodSecurityContext{}
+		}
+		if annotation == profileUnconfined {
+			out.Spec.SecurityContext.SeccompProfile = &v1.SeccompProfile{
+				Type: v1.SeccompProfileUnconfined,
+			}
+		} else if annotation == v1.SeccompProfileRuntimeDefault {
+			out.Spec.SecurityContext.SeccompProfile = &v1.SeccompProfile{
+				Type: v1.SeccompProfileTypeDefault,
+			}
+		} else if strings.HasPrefix(annotation, localhostPrefix) {
+			out.Spec.SecurityContext.SeccompProfile = &v1.SeccompProfile{
+				Type:             v1.SeccompProfileLocalhost,
+				LocalhostProfile: &annotation,
```

Comment:
I think the `annotation` would always be prefixed `localhost/` on this case. However, `LocalhostProfile` needs that removed, as the seccomp type now comes from type discriminator.

Reply:
Yes, changed to trim the prefix.